geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -9990,6 +9996,26 @@ public EntryExpiryTask getEntryExpiryTask(Object key) {
     return br.getEntryExpiryTask(key);
   }
 
+  @Override
+  public void updatePRNodeInformation(CacheLoader cacheLoader, CacheWriter cacheWriter) {
+    Region<String, PartitionRegionConfig> partionedRegionRoot = getPRRoot();
+    if (partionedRegionRoot != null) {
+      PartitionRegionConfig prConfig = partionedRegionRoot.get(getRegionIdentifier());
+      if (prConfig != null) {
+        for (Node node : prConfig.getNodes()) {
+          if (node.getMemberId().equals(getMyId())) {
+            byte loaderByte = (byte) (cacheLoader != null ? 0x01 : 0x00);
+            byte writerByte = (byte) (cacheWriter != null ? 0x02 : 0x00);
+            prConfig.removeNode(node);
+            node.setLoaderWriterByte((byte) (loaderByte + writerByte));
+            prConfig.addNode(node);
+          }
+        }
+        partionedRegionRoot.put(getRegionIdentifier(), prConfig);
```

Comment:
How is this safe if another member is modifying the same PartitionRegionConfig concurrently?
prRoot is a distributed region and two or more members could try to modify the PartitionRegionConfig for the same region identifier. Do we have some locking somewhere to prevent concurrent mutators? In our case we could have sent multiple servers the CreateMappingFunction and they might both be concurrently mutating the region. Seems like we need the prRoot.get and prRoot.put to be done atomically to be safe. 
Also consider using updatePRConfig instead of just calling put.
Dan Smith might know more about this code and how to update the node information safely.

Reply:
Done